google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2423,6 +2471,48 @@ private void releaseSession(PooledSession session, Position position) {
     }
   }
 
+  private boolean isUnbalanced(PooledSession session) {
+    // Don't bother with any randomization if the number of checked out sessions is low, as it is
+    // better to re-use sessions as much as possible in a low-QPS scenario.
+    if (sessions.isEmpty() || checkedOutSessions.size() <= 2) {
+      return false;
+    }
+    int numChannels = sessionClient.getSpanner().getOptions().getNumChannels();
```

Comment:
This line is invoked during the running of the `SessionPoolStressTest`, and is the reason that we need some additional mocking.